src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -1,24 +1,57 @@
-import {NgModule, OnInit, Directive, ElementRef, HostBinding} from '@angular/core';
+import {NgModule, OnInit, Directive, ElementRef, HostBinding, OnDestroy} from '@angular/core';
+import {Event, Router, NavigationEnd, ResolveEnd} from '@angular/router';
+import {filter, map} from 'rxjs/operators';
+import {Subscription} from "rxjs";
 
 /** The timeout id of the previous focus change. */
 let lastTimeoutId = -1;
 
 @Directive({
   selector: '[focusOnNavigation]',
 })
-export class NavigationFocus implements OnInit {
+export class NavigationFocus implements OnInit, OnDestroy {
   @HostBinding('tabindex') role = '-1';
 
-  constructor(private el: ElementRef) {}
+  private softNavFocus = false;
+  private subscriptions = new Subscription();
+
+  constructor(private el: ElementRef, private router: Router) {
+    // subscribing to events in constructor to catch the correct NavigationEnd/ResolveEnd from previous page
+    // and determine whether the element in question needs to be focused
+    // switching between different components/cdk produces a ResolveEnd instead of NavigationEnd event
+    // the focus is set in ngOnInit to account for late loaded content
+    this.subscriptions.add(
+      this.router.events
+        .pipe(
+          filter((event: Event) => event instanceof NavigationEnd || event instanceof ResolveEnd),
+          map(navEnd => navEnd as NavigationEnd))
+      .subscribe((navEnd: NavigationEnd) => {
+        const currentLocation = new URL(window.location.href);
+        if (!currentLocation.hash && isSoftNav(navEnd)) {
+          this.softNavFocus = true;
+          // navigating between CDK/Components pages only runs ngOnInit once so the focus must also be set here
+          this.el.nativeElement.focus({preventScroll: true});
```

Comment:
I thought we were waiting until `ngOnInit` to focus?

Reply:
navigating between the CDK/Components page only runs ngAfterViewInit once because it's the same component (I believe) so we need to track navigating between them through the router events